### PR TITLE
[LibFix] Add option to pass cmd args to create dc with io image

### PR DESCRIPTION
**Pull request by sayaleeraut**
This fix will provide an option to pass a list of arguments
to the image being used.

Signed-off-by: sayaleeraut <saraut@redhat.com>

### CODE REVIEW

**Comment by sayaleeraut** [comment]
Tried running below TCs which use 'oc_create_app_dc_with_io()' after adding changes locally and they ran successfully.

tests/functional/provisioning/test_pv_resize.py::TestPvResizeClass::test_pv_resize_try_shrink_pv_size PASSED
tests/functional/provisioning/test_pv_resize.py::TestPvResizeClass::test_pv_resize_with_prefix_for_name_and_size_1__True__True_ PASSED
tests/functional/provisioning/test_pv_resize.py::TestPvResizeClass::test_pv_resize_with_prefix_for_name_and_size_2__False__True_ PASSED
tests/functional/provisioning/test_pv_resize.py::TestPvResizeClass::test_pv_resize_with_prefix_for_name_and_size_3__False__False_ PASSED

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py [hunk 1]
```diff
@@ -414,7 +417,8 @@ def _oc_create_app_dc_with_io_image(hostname, pvc_name, dc_name_prefix,
 
 def oc_create_app_dc_with_io(
         hostname, pvc_name, dc_name_prefix="autotests-dc-with-app-io",
-        replicas=1, space_to_use=1048576, label=None, image="cirros"):
+        args=None, replicas=1, space_to_use=1048576, label=None,
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: `cmd_args` variable name is more clear to understand

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py [hunk 2]
```diff
@@ -423,6 +427,7 @@ def oc_create_app_dc_with_io(
                         the application PODs where constant I/O will run.
         dc_name_prefix (str): DC name will consist of this prefix and
                               random str.
+        args (list): list of arguments to be passed to the image
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: More clear  description would  be  -
```
list of arguments to be passed to the container command
```

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py (hunk 2 above)
Comment (sayaleeraut) [reply openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: done

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py (hunk 1 above)
Comment (sayaleeraut) [reply openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: done

**Review by vamahaja — APPROVED** [review]

**Review by SushilG96 — APPROVED** [review]
lgtm

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py [hunk 3]
```diff
@@ -423,13 +427,15 @@ def oc_create_app_dc_with_io(
...
+        cmd_args (list): list of arguments to be passed to the container
+                     command
         replicas (int): amount of application POD replicas.
         space_to_use (int): value in bytes which will be used for I/O.
         label (dict): dict of keys and values to add labels in DC.
         image (str): Container image for I/O.
     """
     return _oc_create_app_dc_with_io_image(
-        hostname, pvc_name, dc_name_prefix, replicas, space_to_use,
+        hostname, pvc_name, dc_name_prefix, cmd_args, replicas, space_to_use,
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: `cmd_args` is optional parameter, need to pass correctly as `cmd_args=cmd_args`

**Review by vamahaja — CHANGES_REQUESTED** [review]
Pass `cmd_args` correctly

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py (hunk 3 above)
Comment (sayaleeraut) [reply openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: Done

**Review by balakreddy — APPROVED** [review]
LGTM

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py [hunk 4]
```diff
@@ -352,14 +352,11 @@ def oc_create_pvc(hostname, sc_name=None, pvc_name_prefix="autotests-pvc",
...
+                                    cmd_args=None):
     dc_name = "%s-%s" % (dc_name_prefix, utils.get_random_str())
-    container_data = {
-        "name": dc_name,
-        "image": image,
-        "volumeMounts": [{"mountPath": "/mnt", "name": dc_name}],
-        "command": ["sh"],
-        "args": [
+    if cmd_args is None:
+        args = [
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: This variable should be `cmd_args` as `args` is not used in `container_data`

**Review by vamahaja — CHANGES_REQUESTED** [review]
Function call failing to create io pods as `null` is passed as arguments.

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py (hunk 4 above)
Comment (sayaleeraut) [reply openshift-storage-libs/openshiftstoragelibs/openshift_ops.py]: Done

**Review by vamahaja — APPROVED** [review]

File: openshift-storage-libs/openshiftstoragelibs/openshift_ops.py [hunk 5]
```diff
@@ -423,14 +427,16 @@ def oc_create_app_dc_with_io(
                         the application PODs where constant I/O will run.
         dc_name_prefix (str): DC name will consist of this prefix and
                               random str.
+        cmd_args (list): list of arguments to be passed to the container
```
Comment (srivickynesh) [inline openshift-storage-libs/openshiftstoragelibs/openshift_ops.py:430]: its nice to add an example cmd_args parameters as well here.

**Pull request merged**